### PR TITLE
Navigation Component: Add isText prop to NavigationItem

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
Adds a new prop to `NavigationItem` called `isText`. If set to `true` then rather than acting as a button, `NavigationItem` component will act as a simple non-interactable text. This allows us to show text in the Navigation menu with the same styling as `NavigationItem`, but only acting as a simple text. Great for showing "Loading..." or similar texts for the user.

## How has this been tested?
- `npm run storybook:dev`
- Open `Navigation` -> `Default` story

## Screenshots 
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/navigation/stories/default.js [hunk 1]
```diff
@@ -31,6 +32,15 @@ export function DefaultStory() {
...
 				/>
+				<NavigationItemText
+					item="item-sub-menu-text"
+					title="This is just text, doesn't have any functionality"
+				/>
+				<NavigationItemText
+					item="item-sub-menu-text-badge"
+					title="Text with badge"
+					badge="2"
+				/>
```
Comment (Copons) [inline packages/components/src/navigation/stories/default.js:43]: These should go in `more-examples.js`

**Review by Copons — COMMENTED** [review]
I think there's a simpler way to handle this, by taking advantage of the `ItemBaseUI` styled component's `as` prop.

As far as I can see, the only difference between `NavigationItem` and `NavigationItemText` is that the former renders a `Button`, and the latter a `div` (which is the default).

For example, with an `isText` prop we could do something like this:
```jsx
// Please disregard the very bad code!

function NavigationItem( { isText, ...restProps } ) {
  const itemProps = isText ? { as: Button, href, onClick: onItemClick } : {};

  return <ItemBaseUI { ...{ ...itemProps, ...restProps } } />;
}
```

Optionally we could also expose the additional `NavigationItemText` component to simplify the external interface:
```jsx
function NavigationItemText( props ) {
  return <NavigationItem isText { ...props } />;
}
```

**Review by Addison-Stavlo — COMMENTED** [review]
Nice, I like where this is going and the idea behind it.

Do you think it makes more sense to have to use separate components: `NavigationItem` vs `NavigationItemText` ?  Or to enable different functionality on `NavigationItem` with a separate prop `isText` or `disabled` or something like that?

**Comment by mattwiebe** [comment]
I would tend to echo @Addison-Stavlo’s hesitation re: introducing a whole new component for what seems like a pretty marginal use case?

**Comment by david-szabo97** [comment]
`isText` prop sounds good to me! Updated the implementation. Should we keep the `NavigationBase` and `NavigationBaseContent`? I think they provide a good base in case we need something in the future.

**Review by mattwiebe — APPROVED** [review]
Nice turnaround on the `isText` refactor 👍 

I don't mind the `NavigationBase` / `NavigationBaseContent` stuff but I'm guessing you might not have used them if this hadn't started out with a separate `NavigationItemText` component?

In any case, this looks good to me.

**Comment by Addison-Stavlo** [comment]
> isText prop sounds good to me! Updated the implementation.

Neat, I like it!

> Should we keep the `NavigationBase` and `NavigationBaseContent`?  I think they provide a good base in case we need something in the future.

I don't have a strong opinion either way.  As you say they could prove useful in the future.  I wonder if the codeowners have any thoughts on this.

**Comment by david-szabo97** [comment]
> I don't mind the NavigationBase / NavigationBaseContent stuff but I'm guessing you might not have used them if this hadn't started out with a separate NavigationItemText component?

Yep, I wouldn't have them in this PR. After all these two new internal components provide a good foundation to create new custom item components if needed. On the other hand, `NavigationItem` component is now slimmer so easier to read. Perf-wise it's also better to have separate components to prevent big rerenders. Though we don't have any `memo` here but we might want to optimize later.

File: packages/components/src/navigation/item/index.js [hunk 2]
```diff
@@ -70,34 +64,23 @@ export default function NavigationItem( props ) {
 		onClick( event );
 	};
 	const icon = isRTL ? chevronLeft : chevronRight;
+	const baseProps = isText
+		? { as: 'div', ...restProps }
```
Comment (Copons) [inline packages/components/src/navigation/item/index.js:68]: This `as: 'div'` is unnecessary, given `ItemBaseUI` renders as `div` by default, but I think there's also value in being clear.
No strong opinions about this: feel free to keep it or remove it as you like. 🙂

File: packages/components/src/navigation/item/index.js [hunk 3]
```diff
@@ -70,34 +64,23 @@ export default function NavigationItem( props ) {
...
-
-					{ badge && (
-						<ItemBadgeUI
-							className="components-navigation__item-badge"
-							isRTL={ isRTL }
-						>
-							{ badge }
-						</ItemBadgeUI>
-					) }
+				<ItemBaseUI { ...baseProps }>
```
Comment (Copons) [inline packages/components/src/navigation/item/index.js:74]: I think we should update the styled components names to keep them consistent with where they are used.

For example, we use `ItemBaseUI` inside `NavigationItem` and `ItemUI` inside `NavigationItemBase`.
I understand the reasoning (and its history), but it's also counterintuitive.

Comment (Copons) [reply packages/components/src/navigation/item/index.js:74]: One other thing: what if we moved this `ItemBaseUI` into `NavigationItemBaseContent`?

Something like this:
```jsx
function NavigationItem( props ) {
  const baseContentProps = { title, badge, isRTL };

  const baseProps = isText
    ? { as: 'div', ...baseContentProps, ...restProps }
    : { as: Button, href, onClick: onItemClick, ...baseContentProps, ...restProps };

  return ( <NavigationItemBase>
    <NavigationItemBaseContent { ...baseProps } />
  </NavigationItemBase> );
}

function NavigationItemBaseContent( props ) {
  return ( <ItemBaseUI title={ title } badge={ badge } isRTL={ isRTL }>
    { title && <ItemTitleUI /> }
    { badge && <ItemBadgeUI /> }
  </ItemBaseUI> );
}
```

File: packages/components/src/navigation/item/base-content.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Internal dependencies
+ */
+import { ItemBadgeUI, ItemTitleUI } from '../styles/navigation-styles';
+
+export default function NavigationItemBaseContent( props ) {
+	const { badge, title, isRTL } = props;
```
Comment (Copons) [inline packages/components/src/navigation/item/base-content.js:7]: Should we `useRTL` instead of passing `isRTL` through props?

**Review by Copons — COMMENTED** [review]
The component works great, thanks @david-szabo97! ✨ 

I've left a few comments, they are mostly about making it all more consistent.

**Comment by david-szabo97** [comment]
@Copons Liked your suggestions so I made a few changes. Please give it a peek again, thanks! 🙇

**Review by Copons — APPROVED** [review]

**Pull request merged**